akka-docs/src/main/paradox/scala/camel.md

```diff
@@ -69,68 +95,93 @@ You can also create a CamelMessage yourself with the appropriate body and header
+
+Java
+:  @@snip [ActivationTestBase.java]($code$/java/jdocs/camel/ActivationTestBase.java) { #CamelActivation }
 
 The above code shows that you can get a `Future` to the activation of the route from the endpoint to the actor, or you can wait in a blocking fashion on the activation of the route.
 An `ActivationTimeoutException` is thrown if the endpoint could not be activated within the specified timeout. Deactivation works in a similar fashion:
 
-@@snip [Introduction.scala]($code$/scala/docs/camel/Introduction.scala) { #CamelDeactivation }
+Scala
+:  @@snip [Introduction.scala]($code$/scala/docs/camel/Introduction.scala) { #CamelDeactivation }
+
+Java
+:  @@snip [ActivationTestBase.java]($code$/java/jdocs/camel/ActivationTestBase.java) { #CamelDeactivation }
 
 Deactivation of a Consumer or a Producer actor happens when the actor is terminated. For a Consumer, the route to the actor is stopped. For a Producer, the [SendProcessor](https://svn.apache.org/repos/asf/camel/tags/camel-2.8.0/camel-core/src/main/java/org/apache/camel/processor/SendProcessor.java) is stopped.
 A `DeActivationTimeoutException` is thrown if the associated camel objects could not be deactivated within the specified timeout.
 
 ## Consumer Actors
 
-For objects to receive messages, they must mixin the @extref[Consumer](github:akka-camel/src/main/scala/akka/camel/Consumer.scala)
-trait. For example, the following actor class (Consumer1) implements the
-endpointUri method, which is declared in the Consumer trait, in order to receive
+For objects to receive messages, they must @scala[mixin the @extref[Consumer](github:akka-camel/src/main/scala/akka/camel/Consumer.scala) trait]@java[inherit from the @extref[UntypedConsumerActor](github:akka-camel/src/main/scala/akka/camel/javaapi/UntypedConsumer.scala) class]. 
+For example, the following actor class (Consumer1) implements the
+@scala[`endpointUri`]@java[`getEndpointUri`] method, which is declared in the @scala[`Consumer` trait]@java[@extref[UntypedConsumerActor](github:akka-camel/src/main/scala/akka/camel/javaapi/UntypedConsumer.scala) class], in order to receive
 messages from the `file:data/input/actor` Camel endpoint.
 
-@@snip [Consumers.scala]($code$/scala/docs/camel/Consumers.scala) { #Consumer1 }
+Scala
+:  @@snip [Consumers.scala]($code$/scala/docs/camel/Consumers.scala) { #Consumer1 }
+
+Java
+:  @@snip [Consumer1.java]($code$/java/jdocs/camel/Consumer1.java) { #Consumer1 }
 
 Whenever a file is put into the data/input/actor directory, its content is
 picked up by the Camel [file component](http://camel.apache.org/file2.html) and sent as message to the
 actor. Messages consumed by actors from Camel endpoints are of type
-@extref[CamelMessage](github:akka-camel/src/main/scala/akka/camel/CamelMessage.scala). These are immutable representations of Camel messages.
+[CamelMessage](#camelmessage). These are immutable representations of Camel messages.
```

Comment:
Think it's better to link to a section in the document, as it has more explanation than the code. (in line with Java)